### DISCUSSION TITLE
Creating a custom control

### DISCUSSION THREAD

**Discussion by AlleSchonWeg** [xamarin/XamarinCommunityToolkit#663] [Q&A] [answered]
Hi,
i try to create a custom control/view for my app and i'm not sure what is the correct class to derive from. So i look into xct source code and try to learn. Could you tell me, when i should derive from:

- TemplatedView/BaseTemplatedView 
- ContentView
- View
- Layout<T>

?

I saw that xct uses these 4 baseclasses for creating custom views. But i don't understand why.
Thank you!

**Comment by AndreiMisiukevich** [comment, marked as answer]
Hi
It depends on how would you like to build your view.

1) If you suppose to use custom renderers for your view, then it will make sense to inherit from **View**
2) If your view is kind of layout (The main goal of view is positioning other views inside) then you should inherit from **Layout**
3) If you suppose to build your view 100% cross-platform way (use already existed controls like Expander or RangeSlider etc.) then inherit from **TemplatedView/BaseTemplatedView**

I think so)

**Reply by AlleSchonWeg** [reply]
I need to build a custom control which looks like a table. The control must have a fixed rowheader and a fixed columnheader. Fixed means, that the rowheader must stay visible if the user scrolls horizontally. And vertically the columnheader must stay. Like fixed rows/columns in excel.
The control should take only the data for creating the content(rows, columns, datacells) and internally it must create labels, scrollviews, etc. For consistent width and height of the labels i need measuring. For example the column header label must have the same width of the widest datacell label in that column.

I think for this usecase i should inherit from TemplatedView. 🤔

**Reply by AndreiMisiukevich** [reply]
Probably yes. But you know, it's your code and you are boss there :) so you can inherit from Grid or StackLayout / AbsoluteLayout as I did in CardsView (https://github.com/AndreiMisiukevich/CardView/)
